hudi-sync/hudi-hive-sync/src/main/java/org/apache/hudi/hive/HiveSyncConfig.java

```diff
@@ -98,8 +98,9 @@ public HiveSyncConfig(Properties props) {
 
   public HiveSyncConfig(Properties props, Configuration hadoopConf) {
     super(props, hadoopConf);
-    HiveConf hiveConf = hadoopConf instanceof HiveConf
-        ? (HiveConf) hadoopConf : new HiveConf(hadoopConf, HiveConf.class);
+    HiveConf hiveConf = new HiveConf();
+    // HiveConf needs to load Hadoop conf to allow instantiation via AWSGlueClientFactory
+    hiveConf.addResource(hadoopConf);
```

Comment:
not so sure if this is equivalent to holding the original `hadoopConf` as this changes the order of addResources() during constructing. We should be good only if we can verify the equivalence.

Reply:
i think the ideal approach is to make the passed-in `hiveConf` load hadoop conf properly to use `AWSGlueClientFactory` at the very beginning (when creating hive sync config) so that nothing needs to load at this point.  cc @yihua

Reply:
Loading `AWSGlueClientFactory` property specifically should solve the issue on AWS side, but it's possible that there are other configs/custom configs passed in via Spark session, and won't be loaded by hard-coded logic. I still think loading entire hadoop conf here would be a safer choice.

And this change could change the order of adding resource for hive conf. I've gone thru the `HiveConf` constructor but didn't see any usage of `resource` during constructing so I think it shouldn't affect, maybe I've overlooked something?
An alternative solution would be always pass `hadoopConf` to `HiveConf` constructor. Wdyt?
```suggestion
    HiveConf hiveConf = new HiveConf(hadoopConf, HiveConf.class);
```

Reply:
> but it's possible that there are other configs/custom configs passed in via Spark session,

Is this a classical way people pass around hive options with spark?

> An alternative solution would be always pass hadoopConf to HiveConf constructor

Does it introduce too much overhead then?

Reply:
Hi @danny0405, 
1. It's not the preferred way, but it might be the best way in cases like serverless applications since those can't modify `hive-site.xml` easily
2. There is overhead to load hadoop conf to hive conf instead of casting it directly, but it's almost negligible imo

Reply:
@CTTY always creating a new HiveConf looks good to me.